sui-execution/latest/sui-move-natives/src/crypto/nitro_attestation.rs

```diff
@@ -0,0 +1,129 @@
+    mut args: VecDeque<Value>,
+) -> PartialVMResult<NativeResult> {
+    debug_assert!(ty_args.is_empty());
+    debug_assert!(args.len() == 2);
+
+    let cost = context.gas_used();
+    if !is_supported(context) {
+        return Ok(NativeResult::err(cost, NOT_SUPPORTED_ERROR));
+    }
+
+    let current_timestamp = pop_arg!(args, u64);
+    let attestation_ref = pop_arg!(args, VectorRef);
+    let attestation_bytes = attestation_ref.as_bytes_ref();
+
+    let cost_params = &context
+        .extensions()
+        .get::<NativesCostTable>()
+        .nitro_attestation_cost_params
+        .clone();
+
+    native_charge_gas_early_exit_option!(
+        context,
+        cost_params.parse_base_cost.and_then(|base_cost| cost_params
+            .parse_cost_per_byte
+            .map(|per_byte| base_cost + per_byte * (attestation_bytes.len() as u64).into()))
+    );
+    match parse_nitro_attestation(&attestation_bytes) {
+        Ok((signature, signed_message, payload)) => {
+            let cert_chain_length = payload.get_cert_chain_length();
+            native_charge_gas_early_exit_option!(
+                context,
+                cost_params
+                    .verify_base_cost
+                    .and_then(|base_cost| cost_params
+                        .verify_cost_per_cert
+                        .map(|per_byte| base_cost + per_byte * (cert_chain_length as u64).into()))
+            );
+            match verify_nitro_attestation(&signature, &signed_message, &payload, current_timestamp)
+            {
+                Ok(()) => Ok(NativeResult::ok(
+                    context.gas_used(),
+                    smallvec![Value::struct_(move_vm_types::values::Struct::pack(vec![
+                        Value::vector_u8(payload.module_id.as_bytes().to_vec()),
+                        Value::u64(payload.timestamp),
+                        Value::vector_u8(payload.digest.as_bytes().to_vec()),
+                        Value::vector_for_testing_only(
+                            payload
+                                .pcrs
+                                .iter()
+                                .map(|pcr| Value::vector_u8(pcr.to_vec()))
+                                .collect::<Vec<_>>()
+                        ),
+                        to_option_vector_u8(payload.public_key),
+                        to_option_vector_u8(payload.user_data),
+                        to_option_vector_u8(payload.nonce),
+                    ]))],
+                )),
+                Err(_) => Ok(NativeResult::err(context.gas_used(), VERIFY_ERROR)),
+            }
+        }
+        Err(_) => Ok(NativeResult::err(context.gas_used(), PARSE_ERROR)),
+    }
+}
+
+fn to_option_vector_u8(value: Option<Vec<u8>>) -> Value {
+    match value {
+        Some(vec) => Value::struct_(move_vm_types::values::Struct::pack(vec![Value::vector_u8(
+            vec,
+        )])),
+        None => Value::struct_(move_vm_types::values::Struct::pack(vec![])),
```

Comment:
double check this with @tzakian 

a none prints as 
```
[debug] 0x1::option::Option<vector<u8>> {
}
```
but fails for condition: `assert!(nitro_attestation::get_nonce(&res).is_none());`